cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/ScanApplicationsRequest.java

```diff
@@ -171,6 +186,8 @@ public static class Builder {
     @Nullable
     private NamespaceId namespaceId;
     @Nullable
+    private ApplicationReference applicationReference;
```

Comment:
How is it different from io.cdap.cdap.app.store.ApplicationFilter.ApplicationIdEqualsCaseSensitiveFilter? I feel it's better to push that filter to the database layer

Reply:
I'd put only `String application` here, otherwise we have namespace in two places. The builder can still have a method that takes reference and fills both fields

Reply:
updated